File: 백준/Silver/3036. 링/링.py
import sys
input = sys.stdin.readline

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

N = int(input())
arr = list(map(int, input().split()))
for i in range(1, N):
    m = arr[0] // gcd(arr[0], arr[i])
    c = arr[i] // gcd(arr[0], arr[i])
    print(f'{m}/{c}')

File: 백준/Silver/3036. 링/링.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int [] arr = new int [n];
		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++) {
			int num = Integer.parseInt(st.nextToken());
			arr[i] = num;
		}
		for (int i = 1; i < n; i++) {
			int m = arr[0] / gcd(arr[0], arr[i]);
			int c = arr[i] / gcd(arr[0], arr[i]);
			System.out.println(m+"/"+c);
		}
	}
	public static int gcd(int a, int b) {
		if(b == 0) {
			return a;
		}
		return gcd(b, a%b);
	}
}